dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/EndpointDefinitions.java

```diff
@@ -94,7 +95,24 @@ public EndpointDefinitions(ErrorContext errorContext, Elements elements, Types t
     private Optional<HttpPath> getHttpPath(Element context, Request requestAnnotation) {
         try {
             UriTemplateParser uriTemplateParser = new UriTemplateParser(requestAnnotation.path());
-            return Optional.of(ImmutableHttpPath.of(uriTemplateParser.getNormalizedTemplate()));
+
+            Splitter splitter = Splitter.on('/');
+            Iterable<String> rawSegments = splitter.split(uriTemplateParser.getNormalizedTemplate());
+
+            List<HttpPathSegment> pathSegments = new ArrayList<>();
+            for (String segment : rawSegments) {
+                if (segment.isEmpty()) {
+                    continue; // avoid empty segments; typically the first segment is empty
+                }
+
+                if (segment.startsWith("{") && segment.endsWith("}")) {
+                    pathSegments.add(HttpPathSegments.variable(segment.substring(1, segment.length() - 1)));
```

Comment:
We may want to validate that the more obscure features of UriTemplateParser aren't used, e.g. `/foo/{bar:.*}`. Apologies if we've already validated the template elsewhere.

Reply:
Nope, I haven't. Just copying conjure code for now: I think as you say, we'll want to reimpl this.